core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala

```diff
@@ -986,9 +1026,9 @@ case class TLVPoint(
   }
 }
 
-object TLVPoint extends Factory[TLVPoint] {
+object OldTLVPoint extends Factory[OldTLVPoint] {
 
-  override def fromBytes(bytes: ByteVector): TLVPoint = {
+  override def fromBytes(bytes: ByteVector): OldTLVPoint = {
     val isEndpoint = bytes.head match {
       case 0 => false
       case 1 => true
```

Comment:
ditto

Reply:
i'm not going to touch this code as it is old (`OldTLVPoint`) hopefully we can remove it in the next 3-6 months.

As a separate PR i'm considering something that runs on startup that automatically reads all serialized data structures from the database and writes the new format. After having this for a few releases, we should be able to remove the old data structures all together, at least that is my plan :man_shrugging:

Reply:
i probably won't do this until #3859 is merged though, as the serialization formats included in this PR are still not up to date with the latest DLC protocol.